include/NotificationsManager.php

```diff
@@ -28,109 +32,762 @@ public function __construct() {
+			return $r[0]['total'];
+
+		return 0;
+	}
+
+	/**
+	 * @brief Get introductions
+	 * 
+	 * @param int $all
+	 *	If 0 only include introductions into the query
+	 *	which aren't marked as ignored
+	 * @param int $start Start the query at this point
+	 * @param int $limit Maximum number of query results
+	 * 
+	 * @return array with
+	 *	string 'ident' => Notification identifier
+	 *	int 'total' => Total number of available introductions
+	 *	array 'notifications' => Introductions
+	 */
+	public function introNotifs($all = 0, $start = 0, $limit = 80) {
+		$ident = 'introductions';
+		$total = $this->introTotal($seen);
+		$notifs = array();
+
+		if($all === 0)
+			$sql_extra = " AND `ignore` = 0 ";
+
+		/// @todo Fetch contact details by "get_contact_details_by_url" instead of queries to contact, fcontact and gcontact
+		$r = q("SELECT `intro`.`id` AS `intro_id`, `intro`.*, `contact`.*, `fcontact`.`name` AS `fname`,`fcontact`.`url` AS `furl`,`fcontact`.`photo` AS `fphoto`,`fcontact`.`request` AS `frequest`,
+				`gcontact`.`location` AS `glocation`, `gcontact`.`about` AS `gabout`,
+				`gcontact`.`keywords` AS `gkeywords`, `gcontact`.`gender` AS `ggender`,
+				`gcontact`.`network` AS `gnetwork`
+			FROM `intro`
+				LEFT JOIN `contact` ON `contact`.`id` = `intro`.`contact-id`
+				LEFT JOIN `gcontact` ON `gcontact`.`nurl` = `contact`.`nurl`
+				LEFT JOIN `fcontact` ON `intro`.`fid` = `fcontact`.`id`
+			WHERE `intro`.`uid` = %d $sql_extra AND `intro`.`blocked` = 0
+			LIMIT %d, %d",
+				intval($_SESSION['uid']),
+				intval($start),
+				intval($limit)
+		);
+
+		if(dbm::is_result($r))
+			$notifs = $this->formatIntros($r);
+
+		$arr = array (
+			'ident' => $ident,
+			'total' => $total,
+			'notifications' => $notifs,
+		);
+
+		return $arr;
+	}
+
+	/**
+	 * @brief Format the notification query in an usable array
+	 * 
+	 * @param array $intros The array from the db query
+	 * @return array with the introductions
+	 */
+	private function formatIntros($intros) {
+		$knowyou = '';
+
+		foreach($intros as $it) {
+			// There are two kind of introduction. Contacts suggested by other contacts and normal connection requests.
+			// We have to distinguish between these two because they use different data.
+
+			// Contact suggestions
+			if($it['fid']) {
+
+				$return_addr = bin2hex($this->a->user['nickname'] . '@' . $this->a->get_hostname() . (($this->a->path) ? '/' . $this->a->path : ''));
+
+				$intro = array(
+					'label' => 'friend_suggestion',
+					'notify_type' => t('Friend Suggestion'),
+					'intro_id' => $it['intro_id'],
+					'madeby' => $it['name'],
+					'contact_id' => $it['contact-id'],
+					'photo' => ((x($it,'fphoto')) ? proxy_url($it['fphoto'], false, PROXY_SIZE_SMALL) : "images/person-175.jpg"),
+					'name' => $it['fname'],
+					'url' => zrl($it['furl']),
+					'hidden' => $it['hidden'] == 1,
+					'post_newfriend' => (intval(get_pconfig(local_user(),'system','post_newfriend')) ? '1' : 0),
+
+					'knowyou' => $knowyou,
```

Comment:
you can look into the original code. For friend suggestions the variable `$knowyou` was not defined before. In my opinion `$knowyou` doesn't make any sense for frriend suggestions because the user who is suggesting on of his/her contacts to someone other could not know if they are known to each other.

I just left this inside the code because it was in the original code and I didn't know if there is some meaning from the original author for doing this